7_react/react2/week1/my-app/src/Context/useAppContext.js

```diff
@@ -0,0 +1,46 @@
+import React, { createContext, useContext, useState, useEffect, useCallback } from 'react';
+
+const AppContext = createContext();
+
+const AppProvider = ({ children }) => {
+  const [query, setQuery] = useState('');
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const fetchUsers = useCallback(async () => {

```

Comment:
Good job with the use of useCallback and also the error handling / try-catch block :+1: